### PR TITLE
[AI] Improved handling of old save files

### PR DESCRIPTION
**Pull request by Grummel7**

### CODE REVIEW

File: default/python/AI/freeorion_tools/_freeorion_tools.py [hunk 1]
```diff
@@ -428,25 +428,26 @@ def get_named_real(name: str) -> float:
...
     rules = fo.getGameRules()
     if rules.ruleExistsWithType(name, fo.ruleType.int):
         return rules.getInt(name)
-    error(f"Requested integer rule {name} does not exist!")
-    return 1
+    return default
 
 
-def get_game_rule_real(name: str) -> int:
+def get_game_rule_real(name: str, default: int = 0.0) -> float:
```
Comment (Cjkjvfnby) [inline default/python/AI/freeorion_tools/_freeorion_tools.py]: Should not we preserve the old default value?
```suggestion
def get_game_rule_real(name: str, default: int = 1.0) -> float:
 ```

File: default/python/AI/freeorion_tools/_freeorion_tools.py [hunk 2]
```diff
@@ -428,25 +428,26 @@ def get_named_real(name: str) -> float:
     return 1.0
 
 
-def get_game_rule_int(name: str) -> int:
+def get_game_rule_int(name: str, default: int = 0) -> int:
```
Comment (Cjkjvfnby) [inline default/python/AI/freeorion_tools/_freeorion_tools.py]: Should not we preserve the old default value?
```suggestion
def get_game_rule_int(name: str, default: int = 1) -> int:
```

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Nice catch. 

Looks good, but I have concerns about changing the default value.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Nice catch!

Looks good, but I have concerns about changing the default value.

**Comment by Grummel7** [comment]
> Nice catch!
> 
> Looks good, but I have concerns about changing the default value.

This values can be set when starting a game. When the loading an old save file that does not have them, the games continues with the default values. That's why I modified the AI's default values to match the game's defaults.

Ah, but I see that I missed something: I wanted to remove the function's default, so that the caller always has to figure out the correct default value.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**